Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -432,15 +434,28 @@ void ViewTransition::handleTransitionFrame()
     if (!documentElement)
         return;
 
-    bool hasActiveAnimations = documentElement->hasKeyframeEffects(Style::PseudoElementIdentifier { PseudoId::ViewTransition });
+    auto checkForActiveAnimations = [&](const Style::PseudoElementIdentifier& pseudoElementIdentifier) -> bool {
+        if (!documentElement->animations(pseudoElementIdentifier))
+            return false;
+
+        for (auto& animation : *documentElement->animations(pseudoElementIdentifier)) {
+            if (animation->playState() == WebAnimation::PlayState::Paused || animation->playState() == WebAnimation::PlayState::Running)
```

Comment:
Would be good to only call `animation->playState()` since the work under that function is not necessarily trivial.